charmcraft.yaml

```diff
@@ -1,15 +1,49 @@
 # Copyright 2024 Canonical Ltd.
 # See LICENSE file for licensing details.
-# This file configures Charmcraft.
-# See https://juju.is/docs/sdk/charmcraft-config for guidance.
+
+name: jenkins-agent
+title: Jenkins agent
+summary: Jenkins agent machine charm
+description: |
+  A [Juju](https://juju.is/) [charm](https://juju.is/docs/olm/charmed-operators)
+  deploying and managing [Jenkins](https://www.jenkins.io/) Agent on machines and
+  configurable to use a Jenkins charm deployed in another Juju model.
+
+  This charm simplifies initial deployment and "day N" operations of Jenkins Agent
+  on VMs and bare metal.
+
+  As such, the charm makes it easy for those looking to take control of their own
+  Agents whilst keeping operations simple, and gives them the freedom to deploy on
+  the platform of their choice.
+
+  For DevOps or SRE teams this charm will make operating Jenkins Agent simple and
+  straightforward through Juju's clean interface. It will allow easy deployment
+  into multiple environments for testing changes, and supports scaling out for
+  enterprise deployments.
+links:
+  contact: https://launchpad.net/~canonical-is-devops
+  documentation: https://discourse.charmhub.io/t/jenkins-agent-documentation-overview/12694
+  issues: https://github.com/canonical/jenkins-agent-operator/issues
+  source: https://github.com/canonical/jenkins-agent-operator
+  website: https://charmhub.io/jenkins-agent
 
 type: charm
-bases:
-  - build-on:
-      - name: ubuntu
-        channel: "22.04"
-    run-on:
-      - name: ubuntu
-        channel: "20.04"
-      - name: ubuntu
-        channel: "22.04"
+
+base: ubuntu@24.04
```

Comment:
I think it should support jammy and noble?

Reply:
I think it means that we need multiple charmcraft files?

Reply:
May I know where you might have seen this information?

Reply:
I don't remember where, I have pinged @gregory-schiano to see if he remembers. Otherwise you can ask on the Charmcraft channel for further help

Reply:
Take a look here: https://github.com/canonical/grafana-agent-operator

Reply:
I am referring to the expandable section that refers to from charmcraft version 3 onwards.
I have already asked questions in the charmcraft public channel but havent yet received replies since they are slow.

Reply:
For ci, perhaps we run publish twice and rename the file based on which one we publish? Or we might have to make a change to operator workflows?

Reply:
Yes it looks like it is out of scope for this PR. Perhaps a ticket can be created in the backlog of operator workflows.

Reply:
(We can't run publish twice - we aren't currently able to select which one gets published)

Reply:
that does mean we would need to publish jammy or noble manually for now which seems ok

Reply:
Related: https://github.com/canonical/charmcraft/issues/1789

Reply:
With juju 3.5, this cannot be deployed. noble is currently unsupported.
```
Model  Controller  Cloud/Region         Version  SLA          Timestamp
test   localhost   localhost/localhost  3.5.4    unsupported  15:53:32Z

App            Version  Status   Scale  Charm          Channel  Rev  Exposed  Message
jenkins-agent           blocked      1  jenkins-agent             0  no       Unsupported series.

Unit              Workload  Agent  Machine  Public address  Ports  Message
jenkins-agent/0*  blocked   idle   0        10.248.96.132          Unsupported series.
```

Reply:
@Thanhphan1147 would you mind helping build Jenkins for noble on the PPA please? (or better yet teach me how to do it? where the repository is ...etc)

Reply:
Copied the package from jammy to noble, we should be able to deploy the charm on noble now